### PR TITLE
feat: Add STUN/TURN support

### LINKED ISSUE #111: Enable STUN/TURN clientside

### PR DESCRIPTION
**Pull request by kanyesthaker**
### Description of Changes
Enable STUN/TURN functionality along with logs that verify behavior

### Link to Issue
closes #111 

### Testing Procedure
Verified that we will gather candidates from all available servers. Note that at runtime we may not actually end up gathering all candidates, because simple-peer with trickle ICE will terminate gathering candidates once the first viable candidate has been found.
image

### CODE REVIEW

File: src/network/peerManager.ts [hunk 1]
```diff
@@ -65,8 +67,57 @@ export default class PeerManager {
     this.peer = new SimplePeer({
       initiator: this.isInitiator,
       trickle: true,
+      reconnectTimer: 100,
```
Comment (samjgorman) [inline src/network/peerManager.ts:70]: Any reason this is set to 100ms?

File: src/network/peerManager.ts [hunk 2]
```diff
@@ -86,9 +137,31 @@ export default class PeerManager {
...
+    } else if (candidate.type === 'srflx') {
+      // Indicates candidate was gathered via STUN
+      log.info(`SRFLX (STUN) candidate gathered: ${candidate.candidates}`);
+    } else if (candidate.type === 'relay') {
+      // Indicates candidate was gathered via TURN
+      log.info(`RELAY (TURN) candidate gathered: ${candidate.candidate}`);
+    } else {
+      log.info(
+        `Unknown candidate type ${candidate.type}: ${candidate.candidate}`,
+      );
```
Comment (samjgorman) [inline src/network/peerManager.ts:161]: Very clean here-- nice job

**Review by samjgorman — APPROVED** [review]
Great PR!

File: src/network/peerManager.ts (hunk 1 above)
Comment (kanyesthaker) [reply src/network/peerManager.ts:70]: no, i just found this value somewhere online ¯\_(ツ)_/¯ i should probably just remove this parameter, good eye

**Pull request merged**